Adyen/Core/InstantPaymentComponent.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// Provides a placeholder for payment methods that don't need any payment detail to be filled.
+/// :nodoc:
+public final class InstantPaymentComponent: PaymentComponent {
+
+    /// The ready to submit payment data.
+    public let paymentData: PaymentComponentData?
+
+    /// :nodoc:
+    public let paymentMethod: PaymentMethod
+
+    /// The delegate of the component.
+    public weak var delegate: PaymentComponentDelegate?
+
+    /// :nodoc:
+    public init(paymentMethod: PaymentMethod, paymentData: PaymentComponentData?) {
```

Comment:
do we need to pass `paymentData` to the initializer? Looks like it can be constructed (line 31) from already existing fields. can we remove the field altogether?